src/Network/RequestManager.php

```
@@ -168,47 +168,43 @@
+
+        throw NetworkError::Unexpected->exception(request: $request, response: $response, context: ['%error%' => 'API responded with an unexpected status code: ' . $response->getStatusCode()]);
+    }
+
+    private static function parseError(ResponseInterface $response): string
     {
         $error = '';
 
         try {
             $error = trim((string) $response->getBody());
+            $decoded = json_decode($error, true, 512, JSON_THROW_ON_ERROR);
+            return $decoded ?? $error;
```

PHPStan (2.1.15)

Method OpenFGA\Network\RequestManager::parseError() should return string but returns mixed.


(return.type)